tests/csv_upload_tests.py

```diff
@@ -128,218 +141,123 @@ def test_import_csv_in_new(self):
-        db_name = "csv_into_new_postgres_no_username"
-        table_name = "postgres_no_username"
-        form_data = self.get_full_data(
-            filename, NEW_DATABASE_ID, db_name, table_name, database_flavor=POSTGRES
+    def test_allow_csv_upload_databases(self):
+        db_id = self.get_existing_db_id()
+        assert db_id in [
+            database.id for database in self.importer.allow_csv_upload_databases()
+        ]
+
+    def test_clean_filename(self):
+        original_filename = "foo,+.bar"
+        filename = self.importer._clean_filename(original_filename, "CSV")
+        print(filename)
+        assert filename == "foo.bar"
+        filename = self.importer._clean_filename(filename, "CSV")
+        assert filename == "foo.bar"
+
+    def test_clean_filename_None(self):
+        purpose = "CSV"
+        error_message = "No name is received for {0}".format(purpose)
+        with self.assertRaisesRegex(NameNotAllowed, error_message):
+            self.importer._clean_filename(None, purpose)
+
+    def test_clean_filename_empty(self):
+        filename = "#,'\""
+        purpose = "CSV"
+        error_message = "Name {0} is not allowed for {1}".format(filename, purpose)
+        with self.assertRaisesRegex(NameNotAllowed, error_message):
+            self.importer._clean_filename(filename, purpose)
+
+    def test_convert_database_id(self):
+        database_id = "1"
+        converted_id = self.importer._convert_database_id(database_id)
+        assert converted_id == 1
+
+    def test_convert_database_id_failed(self):
+        database_id = "id"
+        error_message = (
+            "Possible tampering detected, non-numeral character in database-id"
         )
-        conf["POSTGRES_USERNAME"] = ""
-        conf["POSTGRES_PASSWORD"] = POSTGRES
+        with self.assertRaisesRegex(ValueError, error_message):
+            self.importer._convert_database_id(database_id)
+
+    def test_check_table_name(self):
+        table_name = "myNewTableName"
+        assert not self.importer._check_table_name(table_name)
+
+    def test_check_table_name_failed(self):
+        table_name = db.session.query(SqlaTable).first().table_name
+        error_message = "Table name {0} already exists. Please choose another".format(
+            table_name
+        )
+        with self.assertRaisesRegex(NameNotAllowed, error_message):
+            self.importer._check_table_name(table_name)
+
+    def test_create_sqlite_database(self):
+        db_name = "newSqlite"
+        sqlite = "sqlite"
         try:
-            response = self.get_resp(url, data=form_data)
-            message = "No username supplied for PostgreSQL"
-            assert message in response
+            new_database = self.importer._create_database(db_name, sqlite)
+            assert (
+                new_database
+                == db.session.query(models.Database).filter_by(id=new_database.id).one()
+            )
         finally:
-            os.remove(filename)
```

Comment:
is this done implicitly now? otherwise the .db file will still be standing after the tests